ts-rs/src/lib.rs

```diff
@@ -289,13 +292,28 @@ pub trait TS {
         panic!("{} cannot be flattened", Self::name())
     }
 
-    /// Information about types this type depends on.
-    /// This is used for resolving imports when exporting to a file.
-    fn dependencies() -> Vec<Dependency>
-    where
-        Self: 'static;
+    fn dependency_types() -> impl TypeList
+        where
+            Self: 'static {}
+
+    fn dependencies() -> Vec<Dependency> where Self: 'static {
+        use crate::typelist::TypeVisitor;
+
+        let mut deps: Vec<Dependency> = vec![];
+        struct Visit<'a>(&'a mut Vec<Dependency>);
+        impl<'a> TypeVisitor for Visit<'a> {
+            fn visit<T: TS + 'static + ?Sized>(&mut self) {
+                if let Some(dep) = Dependency::from_ty::<T>() {
+                    self.0.push(dep);
+                }
+            }
+        }
+        Self::dependency_types().for_each(&mut Visit(&mut deps));
 
-    /// `true` if this is a transparent type, e.g tuples or a list.  
+        deps
+    }
```

Comment:
I now completely removed the generation for `TS::dependencies()` from the macro crate.
I kept the `dependencies()` function here for now though, with a default implementation (that's not overwritten anywhere), which converts the new `TypeList` into the old `Vec<Dependency>`.